checkstyle.xml

```diff
@@ -0,0 +1,211 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+
+<!--
+    Checkstyle configuration that checks the Google coding conventions from Google Java Style
+    that can be found at https://google.github.io/styleguide/javaguide.html.
+
+    Checkstyle is very configurable. Be sure to read the documentation at
+    http://checkstyle.sf.net (or in your downloaded distribution).
+
+    To completely disable a check, just comment it out or delete it from the file.
+
+    Authors: Max Vetrenko, Ruslan Diachenko, Roman Ivanov.
+ -->
+
+<module name="Checker">
+    <property name="charset" value="UTF-8" />
+
+    <property name="severity" value="error" />
+
+    <!-- Suppress checkstyle for certain files and directories -->
+    <module name="SuppressionFilter">
+        <property name="file" value="checkstyleSuppressions.xml" />
+    </module>
+
+    <property name="fileExtensions" value="java, properties, xml" />
+    <!-- Checks for whitespace                               -->
+    <!-- See http://checkstyle.sf.net/config_whitespace.html -->
+    <module name="FileTabCharacter">
+        <property name="eachLine" value="true" />
+    </module>
+
+    <!--<module name="NewlineAtEndOfFile" />--> <!-- Screwed up on Appveyor for some reason -->
+
+    <module name="TreeWalker">
+        <module name="AvoidNestedBlocks" />
+        <module name="OuterTypeFilename" />
+        <module name="IllegalTokenText">
+            <property name="tokens" value="STRING_LITERAL, CHAR_LITERAL" />
+            <property name="format" value="\\u00(08|09|0(a|A)|0(c|C)|0(d|D)|22|27|5(C|c))|\\(0(10|11|12|14|15|42|47)|134)" />
+            <property name="message" value="Avoid using corresponding octal or Unicode escape." />
+        </module>
+        <module name="AvoidEscapedUnicodeCharacters">
+            <property name="allowEscapesForControlCharacters" value="true" />
+            <property name="allowByTailComment" value="true" />
+            <property name="allowNonPrintableEscapes" value="true" />
+        </module>
+        <module name="LineLength">
+            <property name="max" value="200" />
```

Comment:
First point: that makes sense. I personally prefer longer lines because I have a big screen, but since GitHub limits the line length may as well stick to 100. That'll cause some funky formatting, though.

Second point: That's why this diff is at +4200 -3200 lines changed